lib/saluki-io/src/deser/codec/dogstatsd/mod.rs

```diff
@@ -288,23 +227,67 @@ fn parse_dogstatsd_metric<'a>(
         metric_values.set_timestamp(timestamp);
     }
 
+    Ok((
+        remaining,
+        MetricPacket {
+            metric_name,
+            tags: maybe_tags.unwrap_or_else(TagSplitter::empty),
+            values: metric_values,
+            timestamp: maybe_timestamp,
+            sample_rate: maybe_sample_rate,
+            container_id: maybe_container_id,
+        },
+    ))
+}
+
+pub struct MetricPacket<'a> {
+    pub metric_name: &'a str,
+    pub tags: TagSplitter<'a>,
+    pub values: MetricValues,
+    pub timestamp: Option<u64>,
+    pub sample_rate: Option<f64>,
+    pub container_id: Option<&'a str>,
+}
+
+pub fn build_metric_metadata_from_packet(
+    packet: &MetricPacket<'_>, origin_detection: bool, peer_addr: &ConnectionAddress,
+) -> MetricMetadata {
     let mut metric_metadata = MetricMetadata::default()
-        .with_sample_rate(maybe_sample_rate)
+        .with_sample_rate(packet.sample_rate)
         .with_origin(MetricOrigin::dogstatsd());
 
-    if let Some(container_id) = maybe_container_id {
+    if let Some(container_id) = packet.container_id {
         metric_metadata.origin_entity_mut().set_container_id(container_id);
     }
 
-    Ok((
-        remaining,
-        (
-            metric_name,
-            maybe_tags.unwrap_or_else(TagSplitter::empty),
-            metric_values,
-            metric_metadata,
-        ),
-    ))
+    // We do one optional enrichment step here, which is to add the client's socket credentials as a tag
+    // on each metric, if they came over UDS. This would then be utilized downstream in the pipeline by
+    // origin enrichment, if present.
+    if origin_detection {
```

Comment:
We can actually drop `origin_detection` as a parameter, and this conditional that uses it.

We basically won't end up with any peer address when using UDS _unless_ we configure the socket early on to do grab it... which means that we can't ever get `Some(creds)` unless `origin_detection` is already `true`.

One less conditional, and one less function parameter.

Reply:
Does that apply all the way up to the config option? Seems like that is its only use currently.

Reply:
Went ahead and removed it all the way up.